pandas/util/_test_decorators.py

```diff
@@ -104,8 +104,10 @@ def _skip_if_not_us_locale():
 
 
 def _skip_if_no_scipy():
-    return not (safe_import('scipy.stats') and safe_import('scipy.sparse') and
-                safe_import('scipy.interpolate'))
+    return not (safe_import('scipy.stats') and
+                safe_import('scipy.sparse') and
+                safe_import('scipy.interpolate') and
+                safe_import('scipy.signal'))
```

Comment:
I'm not entirely sure why, but importing here should avoid the test failure. I suspect that whatever modifications pytest makes to the warningsfilter isn't active at this point (in the fixture), so the warning about the dtype size changing isn't elevated to an error.